### DISCUSSION TITLE
How to navigate between multiple screen in a feature module?

### DISCUSSION THREAD

**Discussion by Moes81** [android/nowinandroid#813] [Q&A]
Hi there!
First, I'd like to thank you for this sample app! It seems simple from a functionality point of view, but it provides some helpful examples to solve big team/complexity challenges.

For someone who is doing Android for a while now, but is kind of new to Compose, I'm currently struggling with the navigation stuff. As far as I can see the app has only one screen per feature module. With the provided navigation concept it's fairly easy to navigate between the top level screens. But I have a hard time to figure out how to navigate between multiple screens within a feature module, because I don't have access to the navController within the module.

With the NIA-Navigation concept, how would you solve the following screenflow:
A user can navigate from a home screen to a "Quizz" (that's a no-brainer). The Quizz is a feature module with an 
- introduction screen, 
- (for simplicity sake) 4 hardcoded question screens
- summary screen

Now, I don't see any reason why the `NavHost` should handle the self contained navigation within this feature module. IMHO it should only handle navigation between features, but not navigation within features. Doing the navigation within the `NavGraphBuilder`-Extension functions, e.g. `NavGraphBuilder.interestsGraph` is per se not possible, because we don't have access to the `navController`. 

The question is: How would solve this? Would you pass the `appState` to the navGraph extension functions? Or is there a mechanism I'm not aware of?

Thank you for your support!

**Comment by lihenggui** [comment]
This might help: 
https://github.com/android/nowinandroid/issues/693

https://github.com/android/nowinandroid/pull/701

**Reply by Moes81** [reply]
Hm, how does this answer my question?
#693 is about large screen support and combining different feature modules on one screen
#701 is also about large screens 

I have a self contained screenflow inside one feature module and wonder how you would navigate between those screen without passing down the `navController` or letting the `app` module controll the navigation for a feature module.

**Reply by vhmvd** [reply]
@Moes81 I am exploring a similar use-case. My feature module works something like 

```
Location > Child Location > Child Child Location > Details Of Location
```

What approach did you use to solve this?
